go/kbfs/libkbfs/favorites.go

```diff
@@ -322,18 +370,41 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
 				"Serving possibly stale favorites; new data could not be"+
 					" fetched: %v", err)
 		} else { // Successfully got new favorites from server.
-			oldCache := f.cache
-			f.cache = make(map[Favorite]bool)
+			oldCache := f.favCache
+			f.newCache = make(map[Favorite]favoriteData)
+			f.favCache = make(map[Favorite]favoriteData)
+			f.ignoredCache = make(map[Favorite]favoriteData)
 			f.cacheExpireTime = libkb.ForceWallClock(f.config.Clock().Now()).Add(
 				favoritesCacheExpirationTime)
-			for _, folder := range folders {
-				f.cache[*NewFavoriteFromFolder(folder)] = true
+			for _, folder := range favResult.FavoriteFolders {
+				f.favCache[*NewFavoriteFromFolder(
+					folder)] = favoriteDataFrom(folder)
+			}
+			for _, folder := range favResult.IgnoredFolders {
+				f.ignoredCache[*NewFavoriteFromFolder(
+					folder)] = favoriteDataFrom(folder)
+			}
+			for _, folder := range favResult.NewFolders {
+				f.newCache[*NewFavoriteFromFolder(
+					folder)] = favoriteDataFrom(folder)
 			}
 			session, err := f.config.KBPKI().GetCurrentSession(req.ctx)
 			if err == nil {
 				// Add favorites for the current user, that cannot be deleted.
-				f.cache[Favorite{string(session.Name), tlf.Private}] = true
-				f.cache[Favorite{string(session.Name), tlf.Public}] = true
+				f.favCache[Favorite{string(session.Name), tlf.Private}] = favoriteData{
+					Name:       string(session.Name),
+					FolderType: tlf.Private.FolderType(),
+					// TODO: get the TLF ID
```

Comment:
Maybe you can have special methods that set these IDs and their team IDs from inside the loggedIn method that initializes the disk caches.